streamlit-app/src/team_detector.py

```diff
@@ -0,0 +1,83 @@
+import typing as t
+import numpy as np
+
+from sklearn.cluster import KMeans
+from PIL import Image
+
+
+class TeamDetector():
+    def __init__(self) -> None:
+        pass
+
+    def predict_teams(self, player_imgs: t.List[Image.Image]) -> t.List[int]:
```

Comment:
```suggestion
   @staticmethod
    def predict_teams(player_imgs: list[Image.Image]) -> list[int]:
```
- class has no state, so the method can be static (note: usage may need to be updated to call it as `TeamDetector.predict_teams()`)
- `list` is a standard annotation keyword in Python (no import from typing required)